File: lab5-cookies/src/main/java/login/LogoutServlet.java
package login;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

public class LogoutServlet extends HttpServlet {

    @Override
    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {

        for( var cookie : req.getCookies() ) {

            if (cookie.getName().equals("loginCookie")) {
                cookie.setMaxAge(0);
                resp.addCookie(cookie);
            }

        }

        resp.sendRedirect("/login");

    }
}

File: kotlin/lab05-cookies/login/src/main/java/servlets/LogoutServlet.kt
package servlets

import javax.servlet.http.HttpServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

class LogoutServlet: HttpServlet() {

    override fun doGet(req: HttpServletRequest?, resp: HttpServletResponse?) {

        val cookies = req?.cookies.orEmpty()
        cookies.forEach {

            if (it.name == "jSessionID") {
                it.maxAge = 0
                resp?.addCookie(it)
            }

        }
        resp?.sendRedirect("login")
    }
}

